### PR TITLE
Added new HoloToolkit menu for easy Configuration

### LINKED ISSUE #50: Configuring HoloLens Projects and Scenes is Complex and Cumbersome
I'd like to write two HoloToolkit menu options that automatically apply the settings specified here: 

https://developer.microsoft.com/en-us/windows/holographic/unity_development_overview#Configuring_a_Unity_project_for_HoloLens

I'm thinking something like:

- HoloToolkit
    - Auto Configure Project
    - Auto Configure Scene

### PR DESCRIPTION
**Pull request by jwittner**
Auto Configure -> Scene, applies settings to main camera
Auto Configure -> Project, applies build, quality, and project settings.

See here for applied setting:
https://developer.microsoft.com/en-us/windows/holographic/unity_development_overview#Configuring_a_Unity_project_for_HoloLens

Resolves #50

### CODE REVIEW

**Comment by jwittner** [comment]
Open to lots of comments on these changes as they do some work behind Unity's back to get around limitations in script access to certain functionality.

Also, this probably wants some kind of documentation, but was hoping on clarification on where is best.

Thanks!

File: Assets/HoloToolkit/Utilities/Editor/AutoConfigureMenu.cs [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    /// <summary>
+    /// Configuration options derived from here: 
+    /// https://developer.microsoft.com/en-us/windows/holographic/unity_development_overview#Configuring_a_Unity_project_for_HoloLens
+    /// </summary>
+    public static class AutoConfigureMenu
+    {
+        /// <summary>
+        /// Applies recommended scene settings to the current scenes
+        /// </summary>
+        [MenuItem("HoloToolkit/Auto Configure/Scene", false)]
```
Comment (ForrestTrepte) [inline Assets/HoloToolkit/Utilities/Editor/AutoConfigureMenu.cs]: Someone looking at this menu might not quite know what it is for. Suggestion: "HoloToolkit/Configure/Apply HoloLens Settings to Scene". Too long? Just an idea.

File: Assets/HoloToolkit/Utilities/Editor/AutoConfigureMenu.cs [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+            Camera.main.transform.position = Vector3.zero;
+            Camera.main.clearFlags = CameraClearFlags.SolidColor;
+            Camera.main.backgroundColor = Color.black;
+            Camera.main.nearClipPlane = 0.85f;
+        }
+
+        /// <summary>
+        /// Applies recommended project settings to the current project
+        /// </summary>
+        [MenuItem("HoloToolkit/Auto Configure/Project", false)]
```
Comment (ForrestTrepte) [inline Assets/HoloToolkit/Utilities/Editor/AutoConfigureMenu.cs]: HoloToolkit/Configure/Apply HoloLens ProjectSettings?

File: Assets/HoloToolkit/Utilities/Editor/AutoConfigureMenu.cs [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+            {
+                // NOTE: PlayerSettings.virtualRealitySupported would be ideal, except that it only reports/affects whatever platform tab
+                // is currently selected in the Player settings window. As we don't have code control over what view is selected there
+                // this property is fairly useless from script.
+
+                // NOTE: There is no current way to change the default quality setting from script
+
+                string title = "Updates require text serialization of assets";
+                string message = "Unity doesn't provide apis for updating the default quality or enabling VR support.\n\n" +
+                    "Is it ok if we force text serialization of assets so that we can modify the properties directly?";
```
Comment (ForrestTrepte) [inline Assets/HoloToolkit/Utilities/Editor/AutoConfigureMenu.cs]: This is a well-written error message for the user explaining this situation.

File: Assets/HoloToolkit/Utilities/Editor/AutoConfigureMenu.cs [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+            {
+                // Find the WSA element under the platform quality list and replace it's value with 0
+                string settingsPath = "ProjectSettings/QualitySettings.asset";
+                string matchPattern = @"(m_PerPlatformDefaultQuality.*Windows Store Apps:) (\d+)";
+                string replacePattern = @"$1 0";
+
+                string settings = File.ReadAllText(settingsPath);
+                settings = Regex.Replace(settings, matchPattern, replacePattern, RegexOptions.Singleline);
+
+                //Debug.Log("Updated Quality Settings: " + settings);
```
Comment (ForrestTrepte) [inline Assets/HoloToolkit/Utilities/Editor/AutoConfigureMenu.cs]: Nitpick: Delete this. Or make a nicely-formatted log entry the describes all the changes that were made.

File: Assets/HoloToolkit/Utilities/Editor/ConfigureMenu.cs [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+                string settingsPath = "ProjectSettings/ProjectSettings.asset";
+                string settings = File.ReadAllText(settingsPath);
+
+                // Set VR for WSA to 1
+                string matchPattern = @"(Metro::VR::enable:) (\d+)";
+                string replacePattern = @"$1 1";
+                settings = Regex.Replace(settings, matchPattern, replacePattern, RegexOptions.Singleline);
+
+                // This is a bit more complex, we're looking for the WSA list of enabled VR devices, then
+                // ensuring that the HoloLens is in that list
```
Comment (NeerajW) [inline Assets/HoloToolkit/Utilities/Editor/ConfigureMenu.cs]: HoloLens is actually Windows Holographic as the SDK right?

Comment (jwittner) [inline Assets/HoloToolkit/Utilities/Editor/ConfigureMenu.cs]: In the project settings file it is stored as "HoloLens". That will display in the UI as "Windows Holographic".

**Pull request merged**